client/modules/IDE/components/Editor.jsx

```diff
@@ -264,6 +265,9 @@ class Editor extends React.Component {
         CodeMirror.showHint(this._cm, () => {}, {});
       }
     }
+    if (this.props.vimMode !== prevProps.vimMode) {
+      this._cm.setOption('vimMode', this.props.vimMode);
```

Comment:
The option that we are changing on the CodeMirror instance is the `keyMap`.  Untested, but I think it should be this.  Basically just repeating how you set it up initially but based on the new `vimMode` setting.

```suggestion
      this._cm.setOption('keyMap', this.props.vimMode ? 'vim' : 'sublime');
```

Reply:
Thanks for pointing that out!